pymc4/random_variables/random_variable.py

```diff
@@ -163,15 +163,15 @@ def log_prob(self):
 
 
 class DiscreteRV(RandomVariable):
-    def log_prob(self):
+    def log_prob(self, value):
```

Comment:
I don't think this should change with TF2?

Reply:
Im only halfway certain when I say this but I think we do. Here's my reasoning.

In TF2 because things are eagerly evaluated if you don't pass a value to calculate logp immediately how can eager execution be performed? If I remove the value arg, like I did in my latest commit, we get the exception below in the tests. I also practiced with the code snippet above and if a value isn't passed to logp immediately an exception is raised.

From my understanding in TF1 the graph for logp was created first, and the values were passed latter in `sess.run()` which is why this method didn't need any arguments.

Im still learning TF2 which is why I'm don't want to make any bold statements about correctness at the moment :)
> 
>     def log_prob(self):
>         """Log probability computation.
>     
>         Developer Note
>         --------------
>             Discrete Random Variables are not transformed, unlike continuous
>             Random Variables.
>         """
> >       return self._distribution.log_prob()
> E       TypeError: log_prob() missing 1 required positional argument: 'value'

Reply:
Weird, I can run pymc4 just fine with TF2 without this change, not sure about the test though.

Remember that we implicitly want to support both: default (eager) and `function` (i.e. graph mode, although that one runs inside of eager mode now).

Reply:
Agreed, I just need to figure out how to do that properly and seem to be fumbling a bit. I will continue thinking through it.

Reply:
So this is an error when calling into TFP, did they change their interface by any chance? I think we assume that this just returns a logp tensor, not the actual evaluated logp.

Reply:
Yes the interface seems to have changed. I just tested with TF2 nightly preview and tf-nightly-1.14.1.dev20190316 with tfp-nightly 0.7.0.dev20190316

 In both cases an exception is raised if a value is not provided to `log_prob` Independent verification would be appreciated :)

```
from tensorflow_probability import distributions
print(distributions.Bernoulli(probs=.5).log_prob(1))
log_p = distributions.Bernoulli(probs=.5).log_prob()
```

Reply:
CC @csuter @brianwa84